hs_core/views/resource_access_api.py

```diff
@@ -0,0 +1,182 @@
+            return Response(
+                data={'success': "Resource access privileges added."},
+                status=status.HTTP_202_ACCEPTED
+            )
+
+        if "group_id" in keys and "privilege" in keys:
+            group_to_add = utils.group_from_id(request.data['group_id'])
+            try:
+                user_access.share_resource_with_group(resource,
+                                                      group_to_add,
+                                                      request.data['privilege'])
+                return Response(
+                    data={'success': "Resource access privileges added."},
+                    status=status.HTTP_202_ACCEPTED
+                )
+            except Exception:
+                return Response(
+                    data={'error': "This group may not be added to any resources."},
+                    status=status.HTTP_400_BAD_REQUEST
+                )
+
+        message = "Request must contain a 'resource' ID as well as a 'user_id' or 'group_id'"
+        return Response(
+            data={'error': message},
+            status=status.HTTP_200_OK
+        )
+
+    def delete(self, request, pk):
+        view_utils.authorize(request, pk,
+                             needed_permission=ACTION_TO_AUTHORIZE.EDIT_RESOURCE_ACCESS)
+        keys = request.query_params.keys()
+        user_access = UserAccess(user=request.user)
+        resource = hydroshare.get_resource_by_shortkey(shortkey=pk)
+
+        if "user_id" in keys and "group_id" in keys:
+            message = "Request cannot contain both a 'user_id' and a 'group_id' parameter."
+            return Response(
+                data={'error': message},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        if "user_id" in keys:
+            user_to_remove = utils.user_from_id(request.query_params['user_id'])
+            user_access.unshare_resource_with_user(resource, user_to_remove)
+            return Response(
+                data={'success': "Resource access privileges removed."},
+                status=status.HTTP_202_ACCEPTED
+            )
+
+        if "group_id" in keys:
+            group_to_remove = utils.group_from_id(request.query_params['group_id'])
+            user_access.unshare_resource_with_group(resource, group_to_remove)
+            return Response(
+                data={'success': "Resource access privileges removed."},
+                status=status.HTTP_202_ACCEPTED
+            )
+
+        message = "Request must contain a 'resource' ID as well as a 'user_id' or 'group_id'"
+        return Response(
+            data={'error': message},
+            status=status.HTTP_200_OK
+        )
+
+    def get_serializer_classes(self):
+        return (UserResourcePrivilegeSerializer, GroupResourcePrivilegeSerializer,)
+
+    def get_queryset(self, pk, user):
+        resource = hydroshare.get_resource_by_shortkey(shortkey=pk)
+
+        if resource.user_id == user.id:
```

Comment:
@aphelionz It seems the ```user_id``` is the id of the user who originally created the resource. So we can't use ```user_id``` to check if the requesting user is the owner of the resource. I think the following check should work:
```if user in resource.raccess.owners:```